lib/rules/require-prop-types.js

```diff
@@ -0,0 +1,95 @@
+/**
+ * @fileoverview Prop definitions should be detailed
+ * @author Armano
+ */
+'use strict'
+
+const utils = require('../utils')
+
+function create (context) {
+  // ----------------------------------------------------------------------
+  // Helpers
+  // ----------------------------------------------------------------------
+
+  function objectHasType (nodes) {
+    return !!(nodes
+      .filter(p =>
+        p.key.type === 'Identifier' &&
+        p.key.name === 'type' &&
+        (
+          p.value.type !== 'ArrayExpression' ||
+          p.value.elements.length > 0
+        )
+      )[0])
+  }
+
+  function checkProperties (items) {
+    items.map(cp => {
+      if (cp.type !== 'Property') {
+        return
+      }
+      const key = cp.key.name
+      let hasType = true
+      if (cp.value.type === 'ObjectExpression') { // foo: {
+        hasType = objectHasType(cp.value.properties)
```

Comment:
If this function name is to be accurate you should pass `cp.value` as it's an `ObjectExpression`.